src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/command/flow/ingress/of/IngressFlowModFactory.java

```diff
@@ -72,28 +78,83 @@ public IngressFlowModFactory(
+     * flow).
+     */
+    public OFFlowMod makeSingleVlanForwardMessage(MeterId effectiveMeterId) {
+        FlowEndpoint endpoint = command.getEndpoint();
+        RoutingMetadata metadata = RoutingMetadata.builder()
+                .outerVlanId(endpoint.getOuterVlanId())
+                .build(switchFeatures);
+        OFFlowMod.Builder builder = flowModBuilderFactory
+                .makeBuilder(of, TableId.of(SwitchManager.INGRESS_TABLE_ID), -10)
+                .setMatch(of.buildMatch()
+                        .setExact(MatchField.IN_PORT, OFPort.of(endpoint.getPortNumber()))
+                        .setMasked(MatchField.METADATA,
+                                OFMetadata.of(metadata.getValue()), OFMetadata.of(metadata.getMask()))
+                        .build());
+        return makeForwardMessage(builder, effectiveMeterId, FlowEndpoint.makeVlanStack(endpoint.getInnerVlanId()));
+    }
+
+    /**
+     * Make rule to match inner VLAN tag and forward in in ISL (or out port in case one-switch flow).
+     */
+    public OFFlowMod makeDoubleVlanForwardMessage(MeterId effectiveMeterId) {
+        FlowEndpoint endpoint = command.getEndpoint();
+        RoutingMetadata metadata = RoutingMetadata.builder()
+                .outerVlanId(endpoint.getOuterVlanId())
+                .build(switchFeatures);
+        OFFlowMod.Builder builder = flowModBuilderFactory.makeBuilder(of, TableId.of(SwitchManager.INGRESS_TABLE_ID))
+                .setMatch(of.buildMatch()
+                        .setExact(MatchField.IN_PORT, OFPort.of(endpoint.getPortNumber()))
+                        .setExact(MatchField.VLAN_VID, OFVlanVidMatch.ofVlan(endpoint.getInnerVlanId()))
+                        .setMasked(MatchField.METADATA,
+                                OFMetadata.of(metadata.getValue()), OFMetadata.of(metadata.getMask()))
+                        .build());
+        return makeForwardMessage(builder, effectiveMeterId, FlowEndpoint.makeVlanStack(endpoint.getInnerVlanId()));
     }
 
     /**
      * Make rule to match whole port traffic and route it into ISL/egress end.
      */
-    public OFFlowMod makeDefaultPortFlowMatchAndForwardMessage(MeterId effectiveMeterId) {
-        // FIXME we need some space between match rules (so it should be -10 instead of -1)
-        OFFlowMod.Builder builder = flowModBuilderFactory.makeBuilder(of, TableId.of(SwitchManager.INGRESS_TABLE_ID),
-                                                                      -1)
-                .setCookie(U64.of(command.getCookie().getValue()))
+    public OFFlowMod makeDefaultPortForwardMessage(MeterId effectiveMeterId) {
+        // FIXME we need some space between match rules (so priorityOffset should be -10 instead of -1)
+        OFFlowMod.Builder builder = flowModBuilderFactory
+                .makeBuilder(of, TableId.of(SwitchManager.PRE_INGRESS_TABLE_ID), -1)
```

Comment:
1. Here must be Ingress table.
2. There is a problem with priorities.
Current priorities: 
Double tag 0
Single tag -10
Default Port -1

It means that Default Port will catch all single tag traffic

Default port priority must be changed.

Reply:
yes - fixed